primitives/src/ticker.rs

```diff
@@ -51,34 +51,22 @@ impl AsRef<[u8]> for Ticker {
 }
 
 impl TryFrom<&[u8]> for Ticker {
-    type Error = Error;
+    type Error = ();
 
     fn try_from(s: &[u8]) -> Result<Self, Self::Error> {
         let len = s.len();
-        if len > TICKER_LEN {
-            return Err("ticker too long".into());
-        }
         let mut inner = [0u8; TICKER_LEN];
-        inner[..len].copy_from_slice(s);
-        inner.make_ascii_uppercase();
-        // Check whether the given ticker contains no lowercase characters and return an error
-        // otherwise.
-        if &inner[..len] == s {
+        if len >= TICKER_LEN {
+            // return a ticker with the first TICKER_LEN bytes.
+            inner[..TICKER_LEN].copy_from_slice(&s[..TICKER_LEN]);
             Ok(Ticker(inner))
         } else {
-            Err("lowercase ticker".into())
+            inner[..len].copy_from_slice(s);
+            Ok(Ticker(inner))
         }
     }
```

Comment:
Since this can't fail anymore, it shouldn't use the `TryFrom` trait.

Lets just rename it (maybe `from_slice_truncated`) and make it a normal method.  Also add a doc  comment to say that it is just for tests/benchmarks.